llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -772,7 +773,7 @@ void SIFoldOperands::foldOperand(
   if (UseMI->isRegSequence()) {
     Register RegSeqDstReg = UseMI->getOperand(0).getReg();
     unsigned RegSeqDstSubReg = UseMI->getOperand(UseOpIdx + 1).getImm();
-
+    static std::unordered_set<MachineInstr *> seenMI;
```

Comment:
not sure if function static var is better here or removing const qualifiers from several member functions